cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -176,6 +176,8 @@ LocalTrajectoryBuilder::AddHorizontalRangeData(
     return nullptr;
   }
 
+  // Querying matching and insertion indices must be done here before calling
```

Comment:
Add this comment also in the 3D case, i.e. `mapping_3d/*_local_trajectory_builder.cc`? Similar in other places: keep 3D in sync.